Dockerfile

```diff
@@ -11,43 +11,37 @@ USER 0
 RUN yarn --frozen-lockfile --production --network-timeout 600000
 
 RUN yarn run build
-RUN chown -R node apps libs package.json
-RUN chmod 0400 package.json libs/interfaces/package.json tsconfig.json .prettierrc
-RUN chmod 0500 -R apps libs
 
 ### Production image
 
 FROM $BASE_CONTAINER as app
 
 WORKDIR /app
-
-COPY package.json ./
-COPY --from=builder /src/apps/backend/package.json apps/backend/
-COPY --from=builder /src/apps/frontend/package.json apps/frontend/
-COPY --from=builder /src/libs/interfaces/package.json libs/interfaces/
-COPY --from=builder /src/libs/password-complexity/ libs/password-complexity/
-COPY --from=builder /src/apps/backend/node_modules apps/backend/node_modules
-COPY --from=builder /src/apps/backend/dist apps/backend/dist
-COPY --from=builder /src/dist/ dist/
-COPY --from=builder /src/apps/backend/.sequelizerc /app/apps/backend/
-COPY --from=builder /src/apps/backend/db /app/apps/backend/db
-COPY --from=builder /src/apps/backend/config /app/apps/backend/config
-COPY --from=builder /src/apps/backend/migrations /app/apps/backend/migrations
-COPY --from=builder /src/apps/backend/seeders /app/apps/backend/seeders
-RUN chown node package.json libs libs/interfaces libs/interfaces/package.json libs/password-complexity apps/backend/node_modules apps/backend/seeders apps/backend/config apps/backend/db apps/backend/migrations apps/backend/seeders apps/backend/package.json apps/backend apps/frontend apps/frontend/package.json
-RUN chmod 0400 libs/interfaces/package.json apps/frontend/package.json
+#Note for posterity: the permissions setting is to ensure that the user cannot modify the files accidentally and can't be read, executed or changed by another account.
+#Removing these permissions doesn't improve image size or performance meaningfully, but keeping them adds a potential safeguard.
+COPY --chown= --chmod=0400 package.json ./
+COPY --from=builder --chown= --chmod=0400 /src/apps/frontend/package.json apps/frontend/
+COPY --from=builder --chown= --chmod=0400 /src/libs/interfaces/package.json libs/interfaces/
+COPY --from=builder --chown= --chmod=0500 /src/libs/password-complexity/ libs/password-complexity/
+COPY --from=builder --chown= --chmod=0500 /src/apps/backend/package.json apps/backend/
+COPY --from=builder --chown= --chmod=0500 /src/apps/backend/node_modules apps/backend/node_modules
+COPY --from=builder --chown= --chmod=0500 /src/apps/backend/dist apps/backend/dist
+COPY --from=builder --chown= /src/dist/ dist/
+COPY --from=builder --chown= --chmod=0500 /src/apps/backend/.sequelizerc /app/apps/backend/
+COPY --from=builder --chown= --chmod=0500 /src/apps/backend/db /app/apps/backend/db
+COPY --from=builder --chown= --chmod=0500 /src/apps/backend/config /app/apps/backend/config
+COPY --from=builder --chown= --chmod=0500 /src/apps/backend/migrations /app/apps/backend/migrations
+COPY --from=builder --chown= --chmod=0500 /src/apps/backend/seeders /app/apps/backend/seeders
 
 
 EXPOSE 3000
```

Comment:
expose makes a zero-size layer but still impacts caching for lower layers.  since it doesn't change, let's just throw it above the copies in this stage esp since it'll also allow the copy of the cmd.sh to be closer to the other copies.

Reply:
Good point. I'll make that change.